server/src/controllers/GameControllers/snake.js

```diff
@@ -0,0 +1,229 @@
+      this.head.x += this.xAxisVelocity;
+      this.head.y += this.yAxisVelocity;
+      this.updateCoordinate();
+    }
+  }
+
+  /**
+   * Moves the snakes one steps according to relative change on x axis or y axis.
+   */
+  moveSnakeOneStep() {
+    this.head.x += this.xAxisVelocity; // snake head's new x coordinate
+    this.head.y += this.yAxisVelocity; // snake head's new y coordinate
+    if (this.eatFood(this.head)) {
+      this.updateCoordinate();
+      this.getFoodReward();
+      return true;
+    }
+    if (this.hasCollidedWithItself()
+      || this.hasCollidedWithRival()) {
+      this.lost = true;
+      // @TODO: emit a game mechanics event for game lost;
+      clearInterval(this.interval);
+      return false;
+    }
+    this.updateCoordinate();
+    return true;
+  }
+
+  getFoodReward() {
+    this.length += SNAKE_CONSTANTS.FOOD_LENGTH_REWARD;
+    this.life += SNAKE_CONSTANTS.FOOD_LIFE_REWARD; // increasing the life by some constant unit of seconds second on eating food.
+  }
+
+  /**
+   * updates coordinate in snake'body according to new head position
+   */
+  updateCoordinate() {
+    this.bodyCoordinates.push({
+      x: this.head.x,
+      y: this.head.y,
+    });
+    // removing one tail element to limit the snake length, when snake gets larger than snake length parameter
+    if (this.bodyCoordinates.length > this.length) {
+      this.bodyCoordinates.shift();
+    }
+  }
+
+  /**
+   * Checks collision of the head with it's body
+   */
+  hasCollidedWithItself() {
+    if (this.bodyCoordinates.length > this.length - 1) {
+      const { x, y } = this.getHeadTip();
+      const collided = this.bodyCoordinates.find((collisionPoint) => {
+        if (
+          (x > collisionPoint.x - SNAKE_CONSTANTS.ARC_RADIUS) && (x < collisionPoint.x + SNAKE_CONSTANTS.ARC_RADIUS)
+          && (y > collisionPoint.y - SNAKE_CONSTANTS.ARC_RADIUS) && (y < collisionPoint.y + SNAKE_CONSTANTS.ARC_RADIUS)
+        ) {
+          return true;
+        }
+        return false;
+      });
+      if (collided) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  hasCollidedWithRival() {
+    const { x, y } = this.getHeadTip();
+    const collided = this.rivalBody.find((collisionPoint) => {
+      if (
+        (x > collisionPoint.x - SNAKE_CONSTANTS.ARC_RADIUS) && (x < collisionPoint.x + SNAKE_CONSTANTS.ARC_RADIUS)
+        && (y > collisionPoint.y - SNAKE_CONSTANTS.ARC_RADIUS) && (y < collisionPoint.y + SNAKE_CONSTANTS.ARC_RADIUS)
+      ) {
+        return true;
+      }
+      return false;
+    });
+    if (collided) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * We are drawing arcs on each cooridnate point in the snake body, so to check collision we are getting the most ahead part of the snake head.
+   */
+  // eslint-disable-next-line consistent-return
+  getHeadTip() {
+    const arcRadius = SNAKE_CONSTANTS.ARC_RADIUS;
+    // eslint-disable-next-line default-case
```

Comment:
No harm in adding a default case here I think, then you don't need eslint-disable for default case and consistent-return